lib/apiservers/engine/backends/archive.go

```diff
@@ -461,20 +457,31 @@ func (wm *ArchiveStreamWriterMap) WriterForAsset(proxy proxy.VicArchiveProxy, ci
 }
 
 // Close visits all the archive writer in the trie and closes the actual io.WritCloser
-func (wm *ArchiveStreamWriterMap) Close() {
+func (wm *ArchiveStreamWriterMap) Close() (err error) {
 	defer trace.End(trace.Begin(""))
 
+	numWriter := 0
 	closeStream := func(prefix patricia.Prefix, item patricia.Item) error {
 		if aw, ok := item.(*ArchiveWriter); ok && aw.writer != nil {
 			aw.writer.Close()
 			aw.tarWriter.Close()
 			aw.writer = nil
 			aw.tarWriter = nil
+			numWriter++
 		}
 		return nil
 	}
 
 	wm.prefixTrie.Visit(closeStream)
+
+	// wait for all the streams to finish
+	for i := 0; i < numWriter; i++ {
+		result := <- wm.errchan
+		if result != nil {
+			err = result
+		}
+	}
+	return
```

Comment:
Noting here that we just do a `return` but we do not log the error or return it. Is this intentional?